arch/arm/soc/nxp_kinetis/kl2x/soc.h

```diff
@@ -27,6 +27,10 @@ extern "C" {
 #include <misc/util.h>
 #include <random/rand32.h>
 
+#if defined(CONFIG_SOC_FLASH_MCUX)
+#define FLASH_DRIVER_NAME	FLASH_DEV_NAME
```

Comment:
Seems fine for now, though FYI it looks like @erwango will be doing a rename soon: https://github.com/zephyrproject-rtos/zephyr/issues/5919

Reply:
Indeed, it appears FLASH_DEV_NAME is used directly in number of tests and samples.
hence I don't think it makes sense to keep defining FLASH_DRIVER_NAME which is much less used.
I advise to remove, or will be removed anyway by  #5919.

Reply:
The flash_shell sample didn't work without it. Fine to remove when things get cleaned up.